// @primer/primitives/tokens-next-private/ts/functional/typography/typography.ts
export default {
  fontStack: {
    system: "-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
    sansSerif: "-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
    monospace: "ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace",
  },
  text: {
    display: {
      lineBoxHeight: "3.5rem",
      size: "2.5rem",
      lineHeight: "calc(56/40)",
      weight: 500,
      shorthand: "500 2.5rem/calc(56/40) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
    },
    title: {
      size: {
        large: "2rem",
        medium: "1.25rem",
        small: "1rem",
      },
      lineHeight: {
        large: "calc(48/32)",
        medium: "calc(32/20)",
        small: "calc(24/16)",
      },
      weight: {
        large: 600,
        medium: 600,
        small: 600,
      },
      shorthand: {
        large: "600 2rem/calc(48/32) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
        medium: "600 1.25rem/calc(32/20) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
        small: "600 1rem/calc(24/16) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
      },
    },
    subtitle: {
      size: "1.25rem",
      lineHeight: "calc(32/20)",
      weight: 400,
      shorthand: "400 1.25rem/calc(32/20) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
    },
    body: {
      size: {
        large: "1rem",
        medium: "0.875rem",
        small: "0.75rem",
      },
      lineHeight: {
        large: "calc(24/16)",
        medium: "calc(20/14)",
        small: "calc(20/12)",
      },
      weight: 400,
      shorthand: {
        large: "400 1rem/calc(24/16) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
        medium: "400 0.875rem/calc(20/14) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
        small: "400 0.75rem/calc(20/12) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
      },
    },
    caption: {
      size: "0.75rem",
      lineHeight: "calc(16/12)",
      weight: 400,
      shorthand: "400 0.75rem/calc(16/12) -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'",
    },
    codeBlock: {
      size: "0.8125rem",
      lineHeight: "calc(20/13)",
      weight: 400,
      shorthand: "400 0.8125rem/calc(20/13) ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace",
    },
    codeInline: {
      size: "0.9285em",
      weight: 400,
      shorthand: "400 0.9285em ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace",
    },
  },
};
