// @primer/primitives/tokens-next-private/js/functional/themes/light.js
exports.default = {
  topicTag: {
    borderColor: "#ffffff00",
  },
  highlight: {
    neutral: {
      bgColor: "#fff8c5",
    },
  },
  page: {
    header: {
      bgColor: "#f6f8fa",
    },
  },
  diffBlob: {
    addition: {
      fgColor: {
        text: "#1f2328",
        num: "#1f2328",
      },
      bgColor: {
        num: "#d1f8d9",
        line: "#dafbe1",
        word: "#aceebb",
      },
    },
    deletion: {
      fgColor: {
        text: "#1f2328",
        num: "#1f2328",
      },
      bgColor: {
        num: "#ffcecb",
        line: "#ffebe9",
        word: "#ff818266",
      },
    },
    hunk: {
      bgColor: {
        num: "#54aeff66",
      },
    },
    expander: {
      iconColor: "#636c76",
    },
  },
  diffStat: {
    addition: {
      bgColor: "#1f883d",
    },
  },
  codeMirror: {
    fgColor: "#1f2328",
    bgColor: "#ffffff",
    gutters: {
      bgColor: "#ffffff",
    },
    gutterMarker: {
      fgColor: {
        default: "#ffffff",
        muted: "#636c76",
      },
    },
    lineNumber: {
      fgColor: "#636c76",
    },
    cursor: {
      fgColor: "#1f2328",
    },
    selection: {
      bgColor: "#54aeff66",
    },
    activeline: {
      bgColor: "#eaeef280",
    },
    matchingBracket: {
      fgColor: "#1f2328",
    },
    lines: {
      bgColor: "#ffffff",
    },
    syntax: {
      fgColor: {
        comment: "#24292f",
        constant: "#0550ae",
        entity: "#8250df",
        keyword: "#cf222e",
        storage: "#cf222e",
        string: "#0a3069",
        support: "#0550ae",
        variable: "#953800",
      },
    },
  },
  header: {
    fgColor: {
      default: "#ffffffb3",
      logo: "#ffffff",
    },
    bgColor: "#24292f",
    borderColor: {
      divider: "#57606a",
    },
  },
  headerSearch: {
    bgColor: "#24292f",
    borderColor: "#57606a",
  },
  avatar: {
    bgColor: "#ffffff",
    borderColor: "#d0d7de",
    shadow: "0px 0px 0px 2px #ffffffcc",
  },
  avatarStack: {
    fade: {
      bgColor: {
        default: "#afb8c1",
        muted: "#d0d7de",
      },
    },
  },
  control: {
    bgColor: {
      rest: "#f6f8fa",
      hover: "#eef1f4",
      active: "#e7ebef",
      disabled: "#f6f8fa",
      selected: "#f6f8fa",
    },
    fgColor: {
      rest: "#24292f",
      placeholder: "#6e7781",
      disabled: "#8c959f",
    },
    borderColor: {
      rest: "#d0d7de",
      emphasis: "#868f99",
      disabled: "#d0d7de",
      selected: "#f6f8fa",
      success: "#2da44e",
      danger: "#cf222e",
      warning: "#bf8700",
    },
    iconColor: {
      rest: "#636c76",
    },
    transparent: {
      bgColor: {
        rest: "#00000000",
        hover: "#d0d7de33",
        active: "#d0d7de66",
        disabled: "#f6f8fa",
        selected: "#f6f8fa",
      },
      borderColor: {
        rest: "#00000000",
        hover: "#00000000",
        active: "#00000000",
      },
    },
    danger: {
      fgColor: {
        rest: "#d1242f",
        hover: "#d1242f",
      },
      bgColor: {
        hover: "#ffebe9",
        active: "#ffebe966",
      },
    },
    checked: {
      bgColor: {
        rest: "#0969da",
        hover: "#0860ca",
        active: "#0757ba",
        disabled: "#8c959f",
      },
      fgColor: {
        rest: "#ffffff",
        disabled: "#ffffff",
      },
      borderColor: {
        rest: "#0969da",
        hover: "#0860ca",
        active: "#0757ba",
        disabled: "#8c959f",
      },
    },
  },
  controlTrack: {
    bgColor: {
      rest: "#eaeef2",
      hover: "#dee3e8",
      active: "#d2d8de",
      disabled: "#8c959f",
    },
    fgColor: {
      rest: "#636c76",
      disabled: "#ffffff",
    },
    borderColor: {
      rest: "#00000000",
      disabled: "#8c959f",
    },
  },
  controlKnob: {
    bgColor: {
      rest: "#ffffff",
      disabled: "#f6f8fa",
      checked: "#ffffff",
    },
    borderColor: {
      rest: "#868f99",
      disabled: "#f6f8fa",
      checked: "#0969da",
    },
  },
  counter: {
    borderColor: "#ffffff00",
  },
  button: {
    default: {
      fgColor: {
        rest: "#24292f",
      },
      bgColor: {
        rest: "#f6f8fa",
        hover: "#eef1f4",
        active: "#e7ebef",
        selected: "#e7ebef",
      },
      borderColor: {
        rest: "#d0d7de",
        hover: "#d0d7de",
        active: "#d0d7de",
      },
      shadow: {
        resting: "0px 1px 0px 0px #1f23280a",
        inset: "inset 0px 1px 0px 0px #ffffff40",
      },
    },
    primary: {
      fgColor: {
        rest: "#ffffff",
        disabled: "#ffffffcc",
      },
      iconColor: {
        rest: "#ffffffcc",
      },
      bgColor: {
        rest: "#1f883d",
        hover: "#1c8139",
        active: "#197935",
        disabled: "#95d8a6",
      },
      borderColor: {
        rest: "#1f232826",
        hover: "#1f232826",
        disabled: "#1f232826",
      },
      shadow: {
        selected: "inset 0px 1px 0px 0px #002d114d",
      },
    },
    invisible: {
      fgColor: {
        rest: "#0969da",
        hover: "#0969da",
        disabled: "#8c959f",
      },
      bgColor: {
        rest: "#00000000",
        hover: "#d0d7de33",
        active: "#d0d7de66",
        disabled: "#f6f8fa",
      },
      borderColor: {
        rest: "#00000000",
        hover: "#00000000",
      },
    },
    outline: {
      fgColor: {
        rest: "#0969da",
        hover: "#ffffff",
        active: "#ffffff",
        disabled: "#0969da80",
      },
      bgColor: {
        rest: "#f6f8fa",
        hover: "#0969da",
        active: "#0757ba",
        disabled: "#f6f8fa",
      },
      borderColor: {
        hover: "#d0d7de",
        active: "#d0d7de",
      },
      shadow: {
        selected: "inset 0px 1px 0px 0px #00215533",
      },
    },
    danger: {
      fgColor: {
        rest: "#d1242f",
        hover: "#ffffff",
        active: "#ffffff",
        disabled: "#d1242f80",
      },
      iconColor: {
        rest: "#d1242f",
        hover: "#ffffff",
      },
      bgColor: {
        rest: "#f6f8fa",
        hover: "#a40e26",
        active: "#8b0820",
        disabled: "#f6f8fa",
      },
      borderColor: {
        rest: "#d0d7de",
        hover: "#d0d7de",
        active: "#d0d7de",
      },
      shadow: {
        selected: "inset 0px 1px 0px 0px #4c001433",
      },
    },
  },
  buttonCounter: {
    default: {
      bgColor: {
        rest: "#eaeef2",
      },
    },
    invisible: {
      bgColor: {
        rest: "#eaeef2",
      },
    },
    primary: {
      bgColor: {
        rest: "#002d1133",
      },
    },
    outline: {
      bgColor: {
        rest: "#0969da1a",
        hover: "#ffffff33",
        disabled: "#0969da0d",
      },
      fgColor: {
        rest: "#0550ae",
        hover: "#ffffff",
        disabled: "#0969da80",
      },
    },
    danger: {
      bgColor: {
        hover: "#ffffff33",
        disabled: "#cf222e0d",
        rest: "#cf222e1a",
      },
      fgColor: {
        rest: "#c21c2c",
        hover: "#ffffff",
        disabled: "#d1242f80",
      },
    },
  },
  focus: {
    outlineColor: "#0969da",
    outline: "#0969da solid 2px",
  },
  menu: {
    bgColor: {
      active: "#ffffff00",
    },
  },
  overlay: {
    bgColor: "#ffffff",
    backdrop: {
      bgColor: "#8c959f33",
    },
  },
  selectMenu: {
    borderColor: "#ffffff00",
    bgColor: {
      active: "#b6e3ff",
    },
  },
  sideNav: {
    bgColor: {
      selected: "#ffffff",
    },
  },
  timelineBadge: {
    bgColor: "#eaeef2",
  },
  treeViewItem: {
    leadingVisual: {
      bgColor: {
        rest: "#54aeff",
      },
    },
  },
  underlineNav: {
    borderColor: {
      active: "#fd8c73",
      hover: "#afb8c133",
    },
    iconColor: {
      rest: "#636c76",
    },
  },
  fgColor: {
    default: "#1f2328",
    muted: "#636c76",
    onEmphasis: "#ffffff",
    disabled: "#8c959f",
    link: {
      rest: "#0969da",
    },
    accent: "#0969da",
    success: "#1a7f37",
    attention: "#9a6700",
    severe: "#bc4c00",
    danger: "#d1242f",
    open: "#1a7f37",
    closed: "#d1242f",
    done: "#8250df",
    sponsors: "#bf3989",
  },
  bgColor: {
    default: "#ffffff",
    muted: "#f6f8fa",
    emphasis: "#24292f",
    disabled: "#f6f8fa",
    transparent: "#ffffff00",
    neutral: {
      muted: "#eaeef280",
      emphasis: "#6e7781",
    },
    accent: {
      muted: "#ddf4ff",
      emphasis: "#0969da",
    },
    success: {
      muted: "#dafbe1",
      emphasis: "#1f883d",
    },
    attention: {
      muted: "#fff8c5",
      emphasis: "#9a6700",
    },
    severe: {
      muted: "#fff1e5",
      emphasis: "#bc4c00",
    },
    danger: {
      muted: "#ffebe9",
      emphasis: "#cf222e",
    },
    open: {
      muted: "#dafbe1",
      emphasis: "#1f883d",
    },
    closed: {
      muted: "#ffebe9",
      emphasis: "#cf222e",
    },
    done: {
      muted: "#fbefff",
      emphasis: "#8250df",
    },
    sponsors: {
      muted: "#ffeff7",
      emphasis: "#bf3989",
    },
  },
  borderColor: {
    default: "#d0d7de",
    muted: "#d0d7de66",
    emphasis: "#6e7781",
    disabled: "#d0d7de",
    transparent: "#ffffff00",
    neutral: {
      muted: "#afb8c133",
      emphasis: "#6e7781",
    },
    accent: {
      muted: "#54aeff66",
      emphasis: "#0969da",
    },
    success: {
      muted: "#4ac26b66",
      emphasis: "#2da44e",
    },
    attention: {
      muted: "#d4a72c66",
      emphasis: "#bf8700",
    },
    severe: {
      muted: "#fb8f4466",
      emphasis: "#bc4c00",
    },
    danger: {
      muted: "#ff818266",
      emphasis: "#cf222e",
    },
    open: {
      muted: "#4ac26b66",
      emphasis: "#2da44e",
    },
    closed: {
      muted: "#ff818266",
      emphasis: "#cf222e",
    },
    done: {
      muted: "#c297ff66",
      emphasis: "#8250df",
    },
    sponsors: {
      muted: "#ff80c866",
      emphasis: "#bf3989",
    },
  },
  color: {
    ansi: {
      black: "#24292f",
      "black-bright": "#57606a",
      white: "#6e7781",
      "white-bright": "#8c959f",
      gray: "#6e7781",
      red: "#cf222e",
      "red-bright": "#a40e26",
      green: "#116329",
      "green-bright": "#1a7f37",
      yellow: "#4d2d00",
      "yellow-bright": "#633c01",
      blue: "#0969da",
      "blue-bright": "#218bff",
      magenta: "#8250df",
      "magenta-bright": "#a475f9",
      cyan: "#1b7c83",
      "cyan-bright": "#3192aa",
    },
    prettylights: {
      syntax: {
        comment: "#6e7781",
        constant: "#0550ae",
        "constant-other-reference-link": "#0a3069",
        entity: "#8250df",
        storage: {
          modifier: {
            import: "#24292f",
          },
        },
        "entity-tag": "#0550ae",
        keyword: "#cf222e",
        string: "#0a3069",
        variable: "#953800",
        brackethighlighter: {
          unmatched: "#82071e",
          angle: "#57606a",
        },
        invalid: {
          illegal: {
            text: "#f6f8fa",
            bg: "#82071e",
          },
        },
        carriage: {
          return: {
            text: "#f6f8fa",
            bg: "#cf222e",
          },
        },
        "string-regexp": "#116329",
        markup: {
          list: "#3b2300",
          heading: "#0550ae",
          italic: "#24292f",
          bold: "#24292f",
          deleted: {
            text: "#82071e",
            bg: "#ffebe9",
          },
          inserted: {
            text: "#116329",
            bg: "#dafbe1",
          },
          changed: {
            text: "#953800",
            bg: "#ffd8b5",
          },
          ignored: {
            text: "#eaeef2",
            bg: "#0550ae",
          },
        },
        meta: {
          diff: {
            range: "#8250df",
          },
        },
        sublimelinter: {
          gutter: {
            mark: "#8c959f",
          },
        },
      },
    },
    scale: {
      black: "#1f2328",
      transparent: "#ffffff00",
      white: "#ffffff",
      gray: {
        "0": "#f6f8fa",
        "1": "#eaeef2",
        "2": "#d0d7de",
        "3": "#afb8c1",
        "4": "#8c959f",
        "5": "#6e7781",
        "6": "#57606a",
        "7": "#424a53",
        "8": "#32383f",
        "9": "#24292f",
      },
      blue: {
        "0": "#ddf4ff",
        "1": "#b6e3ff",
        "2": "#80ccff",
        "3": "#54aeff",
        "4": "#218bff",
        "5": "#0969da",
        "6": "#0550ae",
        "7": "#033d8b",
        "8": "#0a3069",
        "9": "#002155",
      },
      green: {
        "0": "#dafbe1",
        "1": "#aceebb",
        "2": "#6fdd8b",
        "3": "#4ac26b",
        "4": "#2da44e",
        "5": "#1a7f37",
        "6": "#116329",
        "7": "#044f1e",
        "8": "#003d16",
        "9": "#002d11",
      },
      yellow: {
        "0": "#fff8c5",
        "1": "#fae17d",
        "2": "#eac54f",
        "3": "#d4a72c",
        "4": "#bf8700",
        "5": "#9a6700",
        "6": "#7d4e00",
        "7": "#633c01",
        "8": "#4d2d00",
        "9": "#3b2300",
      },
      orange: {
        "0": "#fff1e5",
        "1": "#ffd8b5",
        "2": "#ffb77c",
        "3": "#fb8f44",
        "4": "#e16f24",
        "5": "#bc4c00",
        "6": "#953800",
        "7": "#762c00",
        "8": "#5c2200",
        "9": "#471700",
      },
      red: {
        "0": "#ffebe9",
        "1": "#ffcecb",
        "2": "#ffaba8",
        "3": "#ff8182",
        "4": "#fa4549",
        "5": "#cf222e",
        "6": "#a40e26",
        "7": "#82071e",
        "8": "#660018",
        "9": "#4c0014",
      },
      purple: {
        "0": "#fbefff",
        "1": "#ecd8ff",
        "2": "#d8b9ff",
        "3": "#c297ff",
        "4": "#a475f9",
        "5": "#8250df",
        "6": "#6639ba",
        "7": "#512a97",
        "8": "#3e1f79",
        "9": "#2e1461",
      },
      pink: {
        "0": "#ffeff7",
        "1": "#ffd3eb",
        "2": "#ffadda",
        "3": "#ff80c8",
        "4": "#e85aad",
        "5": "#bf3989",
        "6": "#99286e",
        "7": "#772057",
        "8": "#611347",
        "9": "#4d0336",
      },
      coral: {
        "0": "#fff0eb",
        "1": "#ffd6cc",
        "2": "#ffb4a1",
        "3": "#fd8c73",
        "4": "#ec6547",
        "5": "#c4432b",
        "6": "#9e2f1c",
        "7": "#801f0f",
        "8": "#691105",
        "9": "#510901",
      },
    },
  },
  shadow: {
    inset: "inset 0px 1px 0px 0px #1f23280d",
    highlight: "inset 0px 1px 0px 0px #ffffff40",
    resting: {
      xsmall: "0px 1px 0px 0px #1f23280f",
      small: "0px 1px 0px 0px #1f23280a",
      medium: "0px 3px 6px 0px #8c959f26",
    },
    floating: {
      small: "0px 1px 3px 0px #1f23281f, 0px 8px 24px 0px #424a531f",
      medium: "0px 4px 32px -4px #8c959f17, 0px 8px 16px -4px #8c959f17, 0px 24px 48px -12px #8c959f29, 0px 48px 96px -24px #8c959f3d",
      large: "0px 8px 24px 0px #8c959f33",
      xlarge: "0px 12px 28px 0px #8c959f4d",
    },
  },
};
